src/main/groovy/de/tracetronic/jenkins/plugins/ecutestexecution/steps/ProvideLogsStep.groovy

```diff
@@ -0,0 +1,230 @@
+import hudson.FilePath
+import hudson.Launcher
+import hudson.model.Run
+import hudson.model.TaskListener
+import hudson.util.ListBoxModel
+import jenkins.security.MasterToSlaveCallable
+import org.jenkinsci.plugins.workflow.steps.Step
+import org.jenkinsci.plugins.workflow.steps.StepContext
+import org.jenkinsci.plugins.workflow.steps.StepDescriptor
+import org.jenkinsci.plugins.workflow.steps.StepExecution
+import org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution
+import org.kohsuke.stapler.DataBoundConstructor
+
+import java.text.SimpleDateFormat
+import java.util.zip.ZipEntry
+import java.util.zip.ZipInputStream
+
+class ProvideLogsStep extends Step {
+
+    @DataBoundConstructor
+    ProvideLogsStep() {
+        super()
+    }
+
+    @Override
+    StepExecution start(StepContext context) throws Exception {
+        return new Execution(this, context)
+    }
+
+    static class Execution extends SynchronousNonBlockingStepExecution<Void> {
+
+        private static final long serialVersionUID = 1L
+
+        private final transient ProvideLogsStep step
+
+        Execution(ProvideLogsStep step, StepContext context) {
+            super(context)
+            this.step = step
+        }
+        /**
+         * Call the execution of the step in the build and archive returned report logs in jenkins.
+         * Logs a warning if old files are present in report file folder of the workspace.
+         * Clean up report log folder after run.
+         * @return Void
+         */
+        @Override
+        protected Void run() throws Exception {
+            String logDirName = "reportLogs"
+            String logDirPath = PathUtil.makeAbsoluteInPipelineHome(logDirName, context)
+
+            Run<?, ?> run = context.get(Run.class)
+            FilePath workspace = context.get(FilePath.class)
+            Launcher launcher = context.get(Launcher.class)
+            TaskListener listener = context.get(TaskListener.class)
+
+            long startTimeMillis = run.getStartTimeInMillis()
+
+            // Download ecu.test logs to jenkins workspace
+            launcher.getChannel().call(
+                    new ExecutionCallable(startTimeMillis, context.get(EnvVars.class), logDirPath, listener)
+            )
+
+            // Upload ecu.test logs as jenkins artifacts
+            if (workspace.child(logDirName).list().size() > 0) {
+                FilePath[] reportLogs = workspace.list("${logDirName}/**/*.log")
+                def artifactsMap = new HashMap<String, String>()
+                reportLogs.each { log ->
+                    def relativePath = log.getRemote().substring(workspace.getRemote().length() + 1)
+                    artifactsMap.put(relativePath, relativePath)
+                }
+                run.artifactManager.archive(workspace, launcher, listener, artifactsMap)
+                run.addAction(new ProvideLogsAction(run))
+                workspace.child(logDirName).deleteContents()
```

Comment:
check need after changes

Reply:
not sure what is meant here.
this cleans the report folder in the jenkins workspace after archiving, the question is if we need to clean the ws in the first place